### ISSUE TITLE
Discrete distribution fails for array arguments in daipp

### ISSUE THREAD

**Issue by stuhlmueller** [probmods/webppl#470]
I'm looking at the following model using both `dev` and `daipp`:

~~~~{js}
sample(Discrete({ps: [.6, .4]}))
~~~~

It works in dev:
~~~~
$ webppl --version
dev v0.7.0-f24238e /Users/stuhlmueller/Projects/webppl

$ webppl ~/Documents/discrete.wppl
Processing /Users/stuhlmueller/Documents/discrete.wppl

* Program return value:
1
~~~~

It fails in daipp:
~~~~
$ git checkout daipp
Switched to branch 'daipp'
Your branch is up-to-date with 'origin/daipp'.

$ webppl --version
daipp v0.7.0-258bea8 /Users/stuhlmueller/Projects/webppl

$ npm install
> webppl@0.7.0 postinstall /Users/stuhlmueller/Projects/webppl
> ./scripts/adify

$ webppl ~/Documents/discrete.wppl
Processing /Users/stuhlmueller/Documents/discrete.wppl
/Users/stuhlmueller/Projects/webppl/src/util.js:80
  if (xs.length === 0) {
        ^

TypeError: Cannot read property 'length' of undefined
    at Object.sum (/Users/stuhlmueller/Projects/webppl/src/util.js:80:9)
    at discreteSample (/Users/stuhlmueller/Projects/webppl/src/dists.js:1033:25)
    at makeDistributionType.sample (/Users/stuhlmueller/Projects/webppl/src/dists.js:607:16)
    at Object.env.coroutine.sample (/Users/stuhlmueller/Projects/webppl/src/header.js:58:24)
    at env.sample (/Users/stuhlmueller/Projects/webppl/src/header.js:78:26)
    at eval (eval at <anonymous> (/Users/stuhlmueller/Projects/webppl/src/main.js:189:10), <anonymous>:17:24)
    at trampolineRunners.cli (/Users/stuhlmueller/Projects/webppl/src/util.js:32:11)
    at eval (eval at <anonymous> (/Users/stuhlmueller/Projects/webppl/src/main.js:189:10), <anonymous>:5:13)
    at /Users/stuhlmueller/Projects/webppl/src/main.js:189:44
    at Object.timeif (/Users/stuhlmueller/Projects/webppl/src/util.js:232:37)
~~~~

It looks like the cause is that we pass `ad.value(this.params.ps).data` to `discreteSample`, but `ad.value(this.params.ps)` is already a plain array, so doesn't have a `data` parameter.

**Comment by null-a** [comment]
The discrete distribution on daipp takes a vector rather than an array. This should work:

```js
sample(Discrete({ps: Vector([.6, .4])}))
```

**Comment by stuhlmueller** [comment]
Is there a reason not to support plain arrays in this setting, or is this just an incidental restriction that isn't worth addressing because the daipp branch isn't intended for more general use at the moment?

**Comment by null-a** [comment]
No reason. I'm expecting I'll need to change this to support both arrays and vectors before we merge the VI bits from daipp into dev.

**Comment by null-a** [comment]
Fixed by 2ff1d7c7
